### PR TITLE
PLANET-5292 Check EN feature toggle

### PR DESCRIPTION
**Pull request by Inwerpsel**
Ref: https://jira.greenpeace.org/browse/PLANET-5292
Needs to be merged after https://github.com/greenpeace/planet4-master-theme/pull/1160 ! It uses classes introduced in that PR. The acceptance tests for this PR will fail until that is merged anyway.

Check the feature toggle of the EN block for the following cases:
* Whether the block can be used in the editor
* Whether the Engaging Networks admin menu item is displayed.
* Whether you can use the "Progress Bar inside EN Form" style on the counter block.

Because CMB2 interprets a falsy value on a boolean setting as "it should not store a record", it's not possible to set the default value to true, as that would make the field always true. So we need to run a script to turn the feature on everywhere instead. This is done in a post-deploy script (as part of the `p4-run-activator` command). There is a gap between when code is on production and when the post-deploy script runs. So to ensure the block is available before this script has run, we check whether the script has already run, which is stored in the db. That check can be removed in a subsequent deploy.

### CODE REVIEW

File: assets/src/blocks/Counter/CounterBlock.js [hunk 1]
```diff
@@ -81,11 +78,16 @@ export class CounterBlock {
...
         },
-        {
-          name: 'en-forms-bar',
-          label: __( 'Progress Bar inside EN Form', 'planet4-blocks' )
-        }
+
       ]
-    );
+
+    if (!window.p4ge_vars.planet4_options.feature_engaging_networks) {
```
Comment (sagarsdeshmukh) [inline assets/src/blocks/Counter/CounterBlock.js:84]: This condition need to reverse. when EN feature is enabled then only EN form counter block style is available.

**Review by sagarsdeshmukh — COMMENTED** [review]
PR looks good to me 👍  , added a comment about counter block EN form style.

File: classes/controller/menu/class-en-settings-controller.php [hunk 2]
```diff
@@ -18,8 +22,12 @@ class En_Settings_Controller extends Controller {
 	 * Create menu/submenu entry.
 	 */
 	public function create_admin_menu() {
+		// We need to check if the migration already ran, as the EN block is on by default, but we cannot give an option
+		// that was added using CMB2 a default value of 'on', because then it can't be turned off.
+		$migration_ran     = MigrationLog::from_wp_options()->already_ran( M001EnableEnFormFeature::get_id() );
```
Comment (lithrel) [inline classes/controller/menu/class-en-settings-controller.php:27]: I guess we can remove it once this migration passes ? let's keep a post-it or a ticket or something

**Review by lithrel — APPROVED** [review]
:+1:

**Pull request merged**